di-ipv-orchestrator-stub/src/main/java/uk/gov/di/ipv/stub/orc/utils/InheritedIdentityJWTBuilder.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.di.ipv.stub.orc.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JOSEObjectType;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.ECPrivateKey;
+import java.security.spec.EncodedKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.di.ipv.stub.orc.config.OrchestratorConfig.INHERITED_IDENTITY_JWT_ISSUER;
+import static uk.gov.di.ipv.stub.orc.config.OrchestratorConfig.INHERITED_IDENTITY_JWT_SIGNING_KEY;
+import static uk.gov.di.ipv.stub.orc.config.OrchestratorConfig.INHERITED_IDENTITY_JWT_TTL;
+import static uk.gov.di.ipv.stub.orc.config.OrchestratorConfig.INHERITED_IDENTITY_JWT_VTM;
+
+public class InheritedIdentityJWTBuilder {
+
+    public static final String VC_CREDENTIAL_SUBJECT = "credentialSubject";
+    public static final String VOT = "vot";
+    public static final String VTM = "vtm";
+    public static final String VC = "vc";
+    public static final String VC_TYPE = "type";
+    public static final String VERIFIABLE_CREDENTIAL_TYPE = "VerifiableCredential";
+    public static final String INHERITED_IDENTITY_CREDENTIAL_TYPE = "InheritedIdentityCredential";
+    public static final String VC_EVIDENCE = "evidence";
+
+    public static final String EC_ALGO = "EC";
+
+    public static SignedJWT generate(
+            String userId, String vtr, String credentialSubject, String evidence)
+            throws NoSuchAlgorithmException, InvalidKeySpecException, JOSEException {
+        Map<String, Object> vc = new LinkedHashMap<>();
+        vc.put(
+                VC_TYPE,
+                new String[] {VERIFIABLE_CREDENTIAL_TYPE, INHERITED_IDENTITY_CREDENTIAL_TYPE});
+        vc.put(VC_CREDENTIAL_SUBJECT, convertJsonToMap(credentialSubject));
+        vc.put(VC_EVIDENCE, List.of(convertJsonToMap(evidence)));
+        Instant now = Instant.now();
+        JWTClaimsSet claimsSet =
+                new JWTClaimsSet.Builder()
+                        .subject(userId)
+                        .issuer(INHERITED_IDENTITY_JWT_ISSUER)
+                        .notBeforeTime(Date.from(now))
+                        .expirationTime(generateExpirationTime(now))
+                        .claim(VOT, vtr)
```

Comment:
It might be good at some point to make the VOT in the VC not coupled to the requested vtr. That way we can test scenarios where they're requestion a VOT that doesn't match what's in the inherited identity.

Reply:
Valid values("PCL200", "PCL250", "P2") are checked within the list and retrieved, and a warning message is added to the log for the others.

Reply:
Chris, I have a question: If there are more than one occurrence of values 'PCL200', 'PCL250', 'P2' within the VTR array, should we write them as separate VOT array values, rather than within a single VOT entry? Although it may not be feasible from a business perspective, I am considering this possibility based on the data.